dance/dance_pipeline.py

```diff
@@ -145,23 +155,30 @@ def filter(self, relevance_function, output_oeb: Union[str, pathlib.Path] = "fil
             Name of an OEB (Openeye Binary) file for storing the relevant
             molecules. If this file already exists, it will be overwritten!
         """
+        logger.info("Filtering molecules")
+
         self.filter_output_oeb = pathlib.Path(output_oeb)
         self.num_molecules = 0
 
         filtered_molecule_stream = oechem.oemolostream(str(output_oeb))
-        for mol in self._generate_molecules_from_database():
-            if relevance_function(mol):
+        for idx, mol in enumerate(self._generate_molecules_from_database()):
+            if relevance_function(oechem.OEMol(mol)):
```

Comment:
Passing a copy of the molecule to the relevance function, so that the original molecule is not modified.

Reply:
You can also create a copy of the mol (newmol=copy.deepcopy(mol)) to prevent any modifications of the original mol

Reply:
I think `oechem.OEMol(mol)` should create a copy.